internal/plugins/darwin/hostinfo.go

```diff
@@ -0,0 +1,292 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+// +build darwin
+
+package darwin
+
+import (
+	"fmt"
+	"github.com/newrelic/infrastructure-agent/internal/agent"
+	"github.com/newrelic/infrastructure-agent/internal/os/distro"
+	"github.com/newrelic/infrastructure-agent/pkg/entity"
+	"github.com/newrelic/infrastructure-agent/pkg/helpers"
+	"github.com/newrelic/infrastructure-agent/pkg/helpers/fingerprint"
+	"github.com/newrelic/infrastructure-agent/pkg/log"
+	"github.com/newrelic/infrastructure-agent/pkg/plugins/ids"
+	"github.com/newrelic/infrastructure-agent/pkg/sysinfo/cloud"
+	"github.com/shirou/gopsutil/host"
+	"github.com/sirupsen/logrus"
+	"regexp"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	hlog               = log.WithComponent("HostInfoPlugin")
+	kernelVersionRegex = regexp.MustCompile("Version (.*?):")
+)
+
+// HostinfoPlugin returns metadata of the host.
+type HostinfoPlugin struct {
+	agent.PluginCommon
+	cloudHarvester cloud.Harvester // Gather metadata for the cloud instance.
+
+	readDataFromCmd func(cmd string, args ...string) (string, error)
+}
+
+// HostInfoData is the structure returned to the backend by HostinfoPlugin.
+type HostInfoData struct {
```

Comment:
could we extract this struct to be OS independent?

Reply:
We have some small difference between this and linux and also windows one is different. I would say to keep it separately.